jwst/associations/lib/diff.py

```diff
@@ -205,13 +205,12 @@ def compare_asn_lists(left_asns, right_asns):
             diffs.extend(dup_errors)
 
     # Ensure that the product name lists are the same.
-    name_diff = left_product_names ^ right_product_names
-    if name_diff:
-        diffs.append(
-            DifferentProductSetsError(
-                f"Left and right associations do not share a common set of products: {name_diff}"
-            )
-        )
+    left_not_right = sorted(left_product_names - right_product_names)
+    right_not_left = sorted(right_product_names - left_product_names)
+    if left_not_right or right_not_left:
+        left_msg = "Products in left but not right:\n" + "\n\t".join(left_not_right)
+        right_msg = "Products in right but not left:\n" + "\n\t".join(right_not_left)
```

Comment:
I prefer a 4-spaces instead of tab but just a personal preference.
```suggestion
        left_msg = "Products in left but not right:\n" + "\n    ".join(left_not_right)
        right_msg = "Products in right but not left:\n" + "\n    ".join(right_not_left)
```

But that said, this `join` leaves the first entry un-indented, which is probably a bug. Either find an alternate formatting method or just use a loop.